vdom/core.py

```diff
@@ -18,9 +25,9 @@ def to_json(el, schema=None):
     does not match the schema.
     """
     if (type(el) is str):
-        return el
-    if (type(el) is list):
-        return list(map(toJSON, el))
+        json_el = el
+    elif (type(el) is list):
+        json_el = list(map(to_json, el))
```

Comment:
I'm assuming you moved this into being validated for purity sake. The reason we were short circuiting it before was that it was already valid (since there was no schema, it was the primary invariant here). I think this is fine to send through validation...

Reply:
Now that it dawns on me, pure strings and arrays are _not_ valid vdom by themselves (must be a child element). We explicitly have to have an object at the top level due to Jupyter JSON mimetype limitations.

Reply:
It would not be hard to wrap them in a dict if we had a programmatic way of thinking about that… what kinds of strings and Arrays should be allowed? How should they be transformed to objects?

Reply:
The top level string could be in a `span`, `div`, `p`, `code`, etc. if we were opinionated on it (note that has an important distinction for layout). An array could be wrapped in a `div`.

However, React handles both `string`s and `Arrays` as top level elements. I'd rather we wait on the next major bump in both the message protocol and notebook format, allowing arrays or strings at the top level.